pandas/core/arrays/datetimes.py

```diff
@@ -1907,6 +1912,20 @@ def std(
 # Constructor Helpers
 
 
+@overload
+def sequence_to_datetimes(
+    data, allow_object: Literal[False] = ..., require_iso8601: bool = ...
+) -> DatetimeArray:
+    ...
+
+
+@overload
+def sequence_to_datetimes(
+    data, allow_object: Literal[True] = ..., require_iso8601: bool = ...
+) -> Union[np.ndarray, DatetimeArray]:
+    ...
+
+
```

Comment:
with these Literal[True]/Literal[False] may also need to add an extra bool case. ie. duplicate the function signature as an additional overload (follow-on ok as may not be needed if this is internal) see #40200 and #40197

Reply:
> may not be needed if this is internal

yah this shouldnt be user-facing